ocean_lib/models/datatoken_base.py

```diff
@@ -338,6 +350,22 @@ def create_dispenser(self, tx_dict: dict, *args, **kwargs):
 
         return tx
 
+    @enforce_types
+    def get_dispensers(self, active=True) -> list:
+        """return List[Dispenser] - all the dispensers for this datatoken"""
+        # import here to avoid circular import
+        from ocean_lib.models.dispenser import Dispenser
+
+        dispensers = []
+        addrs = self.getDispensers()
+        for address in addrs:
+            dispenser = Dispenser(self.config_dict, address)
+            is_active = dispenser.status(self.address)
+            if active and not is_active:
+                continue
+            dispensers.append(dispenser)
```

Comment:
FYI, `dispenser.status()` returns a tuple. Therefore the line `is_active = dispenser.status(self.address)` has a bug. Better is to call `dispenser_status()` to get a `DispenserStatus` object.

Here's a fix to the bug, where code is cleaner too. 
```python
            if active and self.dispenser_status().active:
                dispensers.append(dispenser)
```